src/source_sender/mod.rs

```diff
@@ -276,4 +321,125 @@ impl Inner {
 
         Ok(())
     }
+
+    /// Calculate the difference between the reference time and the
+    /// timestamp stored in the given event reference, and emit the
+    /// different, as expressed in milliseconds, as a histogram.
+    fn emit_lag_time(&self, event: EventRef<'_>, reference: i64) {
+        if let Some(lag_time_metric) = &self.lag_time {
+            let timestamp = match event {
```

Comment:
There are some formats that don't require an event timestamp, OTLP for example. We probably need to set a standard that we'd insert `now()` for missing timestamps.

Reply:
Wouldn't that be an unrelated change, possibly related to schemas, as it changes the data model for those sources?

Reply:
Yeah, probably? I can't think of many other places where we're relying on the timestamp to be there (aside from sinks), just wanted to make sure this won't blow up if the field doesn't exist.